src/services/NumberService.ts

```diff
@@ -133,14 +133,46 @@ export class NumberService {
                     .catch((error: AWSError) => {
                         console.error(error); // limit to 5 attempts
                         if (error.statusCode === 400) {
-                            console.error(`Attempt number ${attempts} failed. Retrying up to ${Configuration.getInstance().getMaxAttempts()} attempts.`);
+                            console.error(`Attempt number ${attempts} for systemNumber failed. Retrying up to ${Configuration.getInstance().getMaxAttempts()} attempts.`);
                             return this.createSystemNumber(attempts + 1, error);
                         }
                         throw this.formatAWSError(error);
                     });
             });
     }
 
+    /**
+     * Creates a new plate serial number in the database.
+     * @param attempt - the current number of attempts for generating a Plate Serial Number.
+     * @param awsError - AWS error to be passed when the request fails otherwise null.
+     */
+    public createPlateSerialNumber(attempts: number, awsError: AWSError | null): Promise<PlateSerialNumber> {
```

Comment:
Some pseudo that maybe we could look at together.
We could either use async or chain the promises by 'flattening' them as we seem to be doing 2 things
getLastPlateSerialNumber() & transactWrite().

These could be separate promises that are passed in the promise chain when calling createPlateSerialNumber()
```
// one promise performing stuff
const promise1 = (payload) => {
  ... we do stuff with payload here
  return Promise.resolve(payload) // then pass it a long or give what is needed for the next promise to be chained 
}

const promise2 = (payload) => {
  ... we do more stuff here with the payload and start composing promises
  return Promise.resolve(payload) 
}

// now call the main promise and chain operations
createPlateSerialNumber()
  .then(promise1)
  .then(promise2)
  .catch(e => throw e)

```
I think this could work and would flatten the calls.